azero/contracts/most/lib.rs

```diff
@@ -276,10 +279,18 @@ pub mod most {
             self.data.set(&data);
 
             // return surplus if any
-            if let Some(surplus) = base_fee.checked_sub(current_base_fee) {
+            if let Some(surplus) = transferred_fee.checked_sub(current_base_fee) {
                 self.env().transfer(sender, surplus)?;
             };
 
+            self.env().emit_event(CrosschainTransferRequest {
+                committee_id: data.committee_id,
+                dest_token_address,
+                amount,
+                dest_receiver_address,
+                request_nonce: data.request_nonce.saturating_sub(1),
```

Comment:
why subtracting 1 here?

Reply:
changed the ordering so that emission is at the end - request_nonce has already been updated

Reply:
then I suggest:
- get the nonce from storage (it has copy semantics)
- update it in the storage, use checked math
- use original value in the event emission.

cheaper in gas and less confusing to read